### DISCUSSION TITLE
Is ApkTool lossless or not? Can I use it to modify my *own* apk and will it introduce bugs?

### DISCUSSION THREAD

**Discussion by fzyzcjy** [iBotPeaches/Apktool#2730] [Q&A] [answered]
Hi thank you so much for the lib! I wonder whether it is lossless or not? I have my own project and want to modify an already generated `.apk` file. (Indeed I am trying to solve [this](https://stackoverflow.com/questions/70538440/how-to-build-a-single-apk-not-apks-but-is-splitted-by-abi-from-aab) using [this](https://platinmods.com/threads/how-to-turn-a-split-apk-into-a-normal-non-split-apk.76683/) but that is kind of unrelated to the question.) 

I am worried: If I use `apktool d` + modifying it + `apktool b`, will I introduce some bugs to my app? Or will it be completely ok and lossless?

**Comment by iBotPeaches** [comment, marked as answer]
I've moved this to discussions. Apktool is NOT loseless. It can and possibly might introduce bugs, there is just a nature of disassembling source and reassembling (smali) and breaking down `resources.arsc` and then rebuilding with aapt2 may lead to issues. Thats the nature of RE.

Though, I did see in history one major company Niantic did use Apktool to once release a version of Ingress on a different package name - https://connortumbleson.com/2019/06/02/apktool-in-the-wild/

**Reply by fzyzcjy** [reply]
Thank you!
